mm2src/coins/utxo/rpc_clients/electrum_rpc/client.rs

```diff
@@ -934,9 +939,24 @@ impl UtxoRpcClientOps for ElectrumClient {
         let mut hashes = vec![hex::encode(electrum_script_hash(&output_script))];
 
         // If the plain pubkey is available, fetch the balance found in P2PK output as well (if any).
+        // The output script (scriptPubkey) of such utxos may have the full pubkey length of 65 bytes or the compressed
+        // version of it (33 bytes). Both are valid and accepted by the blockchain.
         if let Some(pubkey) = address.pubkey() {
-            let p2pk_output_script = output_script_p2pk(pubkey);
-            hashes.push(hex::encode(electrum_script_hash(&p2pk_output_script)));
+            let pubkey = try_f!(pubkey.to_secp256k1_pubkey().map_err(|err| {
+                JsonRpcError::new(
+                    UtxoJsonRpcClientInfo::client_info(self),
+                    rpc_req!(self, "blockchain.scripthash.get_balance").into(),
+                    JsonRpcErrorType::Internal(err.to_string()),
+                )
+            }));
+            let compressed_pubkey = keys::Public::Compressed(pubkey.serialize().into());
```

Comment:
There is code duplication in getting pubkeys from an address, here and in the list_unspent fn.
I guess you could make a function like `fn all_output_scripts(address: Address) -> Result<Vec<Script>, ...> {}`

Reply:
thanks
done here: [4645f9b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2410/commits/4645f9b63610b6e05f4555334bc3a5fdc6413122)

Reply:
Great, thank you.

I also think it would be good to make functions like output_script and all_output_scripts as a member of a struct. Apparently we can't make them as part of the Address struct (bcs mutual refs would appear for Script and Address) so we could create a new type like Destination (and have those functions in its impl) and use it instead of Address. This could be done as part of refactoring ofc.

Reply:
> so we could create a new type like Destination and use it instead of Address

didn't get that. do u want such a type to just host these methods or be used instead of `Address` all together? which stuff such types would be used in?

i feel we can host such methods in `Script`, so `Script::output_script(&addr)` etc... would be fine.

im neutral on this anyways so i will pick whatever u think good.

Reply:
Yes I was thinking to have a higher level type like a Destination type which could depend on the Script type (bcz the Address type could not), so we could have standalone functions like output_script as methods of such Destination type.
But looks like this is not good to have both Destination and Address type, so this is not worth it.